src/services/codefixes/fixEnableExperimentalDecorators.ts

```diff
@@ -0,0 +1,60 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "enableExperimentalDecorators";
+    const errorCodes = [
+        Diagnostics.Experimental_support_for_decorators_is_a_feature_that_is_subject_to_change_in_a_future_release_Set_the_experimentalDecorators_option_to_remove_this_warning.code
+    ];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions: (context) => {
+            const { configFile } = context.program.getCompilerOptions();
+            if (configFile === undefined) {
+                return undefined;
+            }
+
+            const changes = textChanges.ChangeTracker.with(context, changeTracker => makeChange(changeTracker, configFile));
+            return [createCodeFixActionNoFixId(fixId, changes, Diagnostics.Enable_the_experimentalDecorators_option_in_your_configuration_file)];
+        },
+        fixIds: [fixId],
+    });
+
+    function makeChange(changeTracker: textChanges.ChangeTracker, configFile: TsConfigSourceFile) {
+        const tsconfigObjectLiteral = getTsConfigObjectLiteralExpression(configFile);

```

Comment:
Probably adding option to `complerOptions` of the tsconfig can be pulled out to a function and hence repeating is not needed. I see similar code in `fixCannotFindModule.ts`

Reply:
I've extracted the changes & made them more general in `helpers.ts` but I'm not seeing a good way to reduce the amount of code in `fixCannotFindModule.ts` 😕. It does a bit more logic than just setting a value so I'm not seeing how to pull some of it out. Do you have any pointers?

Reply:
Yeah realise that its not very fn sharing friendly. Sorry about that.

Reply:
😄 thanks for the quick review!